plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/CabalAdd.hs

```diff
@@ -204,12 +204,16 @@ hiddenPackageSuggestion diag = getMatch (msg =~ regex)
     msg :: T.Text
     msg = _message diag
     regex :: T.Text -- TODO: Support multiple packages suggestion
-    regex = "It is a member of the hidden package [\8216']([a-zA-Z0-9-]*[a-zA-Z0-9])(-([0-9\\.]*))?[\8217']"
+    regex =
+        "It is a member of the hidden package [\8216']([a-zA-Z0-9-]*[a-zA-Z0-9])(-([0-9\\.]*))?[\8217']"
+        <> "|"
+        <> "needs flag -package-id ([a-zA-Z0-9-]*[a-zA-Z0-9])(-([0-9\\.]*))?"
     -- Have to do this matching because `Regex.TDFA` doesn't(?) support
     -- not-capturing groups like (?:message)
     getMatch :: (T.Text, T.Text, T.Text, [T.Text]) -> [(T.Text, T.Text)]
     getMatch (_, _, _, []) = []
-    getMatch (_, _, _, [dependency, _, cleanVersion]) = [(dependency, cleanVersion)]
+    getMatch (_, _, _, [dependency, _, cleanVersion, "", "", ""]) = [(dependency, cleanVersion)]
+    getMatch (_, _, _, ["", "", "", dependency, _, cleanVersion]) = [(dependency, cleanVersion)]
     getMatch (_, _, _, _) = error "Impossible pattern matching case"
```

Comment:
I know you didn't implement this, but I would feel much better if this `error` wasn't there :) Could you fix this while you are here?

I think we can simply ignore this case, e.g. `getMatch (_, _, _, _) = []`.